src/main/java/com/puppycrawl/tools/checkstyle/filters/PatchXpathFilterElement.java

```diff
@@ -296,9 +330,16 @@ else if (lineNo > childAstLineNoMap.get(MAX)) {
         }
     }
 
-    private boolean isMatchingAst(DetailAST root, TreeWalkerAuditEvent event) {
-        return root.getType() == event.getTokenType()
-                && root.getLineNo() == event.getLine()
-                && root.getColumnNo() == event.getColumnCharIndex();
+    /**
+     * Check whether AST node matches event's node.
+     *
+     * @param ast DetailAST
+     * @param event {@code TreeWalkerAuditEvent} object
+     * @return true if it is matching.
+     */
+    private boolean isMatchingAst(DetailAST ast, TreeWalkerAuditEvent event) {
```

Comment:
to indicate that a method doesn't access instance fields or methods. This will guarantee that static method does not change `this` object state.
(+ some more reasons in https://stackoverflow.com/questions/11240178/what-is-the-gain-from-declaring-a-method-as-static)

Reply:
@baratali thanks, I got it, and according to this principle, `getEventAst`, `getCheckShortName`, `containsShortName`. also can be static, do we need to change them to static?